framework/include/utils/RankTwoTensorImplementation.h

```diff
@@ -146,7 +146,8 @@ void
 RankTwoTensorTempl<T>::fillFromInputVector(const std::vector<T> & input, FillMethod fill_method)
 {
   if (fill_method != autodetect && fill_method != input.size())
-    mooseError("Expected an input vector size of ", fill_method, " to fill the RankTwoTensorTempl");
+    mooseException(
+        "Expected an input vector size of ", fill_method, " to fill the RankTwoTensorTempl");
```

Comment:
I am not sure if we should change mooseError to mooseException. For your code, I think you will throw an error at the end, but if others call this function somewhere else, only exception will be generated. Let us bring this issue to our standup meeting tomorrow.

Reply:
According to the discussion on slack, it sounds like the `mooseException` would error the same way as `mooseError`, if we do not catch it - Even though it does not seem to break things now, I agree it is better to bring this up for a broader discussion before we move on :)